src/snap/metric.cc

```diff
@@ -263,3 +276,91 @@ void Metric::Metric::set_last_advert_tm(system_clock::time_point tp) {
     tm->set_sec(nanos / std::nano::den);
     tm->set_nsec(nanos % std::nano::den);
 }
+
+
+
+Namespace::Namespace(std::vector<std::string> ns){
+    
+    for (auto &string_iterator : ns){
+        this->namespace_elements.push_back(NamespaceElement(string_iterator));
+    }
+}
+Namespace::Namespace(){}
+
+Namespace::~Namespace(){
+
```

Comment:
Unnecessary empty line, please write a destructor in the same style as a constructor.